lib/membrane_dashboard_web/live/dashboard_live.ex

```diff
@@ -84,74 +84,97 @@ defmodule Membrane.DashboardWeb.DashboardLive do
     end
   end
 
   # inits, realoads or updates charts and reloads dagre
   @impl true
-  def handle_info({:charts_init, data}, socket),
-    do: {:noreply, push_event(socket, "charts_init", %{data: data})}
+  def handle_info({:charts_init, data}, socket) do
+    {:noreply, push_event(socket, "charts_init", %{data: data})}
+  end
+
+  def handle_info({event, charts, data, paths}, socket)
+      when event in [:charts_data, :charts_update] do
+    socket =
+      socket
+      |> push_event(Atom.to_string(event), %{data: charts})
+      |> assign(data: data, paths: paths)
+
+    {:noreply, socket}
+  end
+
+  def handle_info({:alive_pipelines, alive_pipelines}, socket) do
+    {:noreply, assign(socket, alive_pipelines: alive_pipelines)}
+  end
+
+  # this message is the last message sent from query task
+  def handle_info({:update_query_time, time_from, time_to}, socket) do
+    socket =
+      socket
+      |> assign(time_from: time_from, time_to: time_to)
+      |> case do
+        # if the timer had not been set then do nothing
+        %{assigns: %{update_ref: nil}} = socket ->
+          socket
 
-  def handle_info({:charts_data, data}, socket),
-    do: {:noreply, push_event(socket, "charts_data", %{data: data})}
+        socket ->
+          socket |> plan_update()
+      end
 
-  def handle_info({:charts_update, data}, socket),
-    do: {:noreply, push_event(socket, "charts_update", %{data: data})}
+    {:noreply, socket}
+  end
 
-  def handle_info({:dagre_data, data}, socket),
-    do: {:noreply, push_event(socket, "dagre_data", %{data: data})}
+  def handle_info({:dagre_data, data}, socket) do
+    {:noreply, push_event(socket, "dagre_data", %{data: data})}
+  end
+
+  def handle_info({:set_data_loading, loading}, socket) do
+    {:noreply, assign(socket, data_loading: loading)}
+  end
 
   def handle_info(:update, socket) do
-    if socket.assigns.update and socket.assigns.next_update_time <= now() do
-      socket = assign(socket, next_update_time: now(@min_between_updates_interval))
+    {:noreply,
+     push_patch_with_params(socket, %{
+       mode: :update,
+       from: now(-socket.assigns.update_range),
+       to: now()
+     })}
+  end
 
-      {:noreply,
-       push_patch_with_params(socket, %{
-         mode: :update,
-         from: now(-socket.assigns.update_range),
-         to: now()
-       })}
-    else
-      {:noreply, socket}
-    end
+  def handle_info(
+        {:DOWN, query_task_ref, :process, _pid, reason},
+        %{assigns: %{query_task_ref: query_task_ref}} = socket
+      ) do
+    {:noreply, assign(socket, query_task_ref: nil)}
   end
 
   @impl true
   def handle_event("refresh", %{"timeFrom" => time_from, "timeTo" => time_to}, socket) do
     with {:ok, {from, to}} <- parse_time_range(time_from, time_to) do
+      # FIXME: why the hell there is update false?
       {:noreply, push_patch_with_params(socket, %{from: from, to: to, update: false})}
     else
+      # this flash is rather useless
```

Comment:
So it can be change to something different but I think it is good to tell user that time range is invalid.

Reply:
At the time I thought that the html input would not allow to put invalid values there. It happens that it can... so in this case this information is rather useful.